src/Collections/MemoryConfigCollection.php

```diff
@@ -76,7 +75,7 @@ public function transform($transformers)
         return $this;
     }
 
-    public function set($class, $name, $data, $metadata = [])
+    public function set(string $class, string|null $name, mixed $data, array $metadata = []): static
```

Comment:
the `string|null` $name isn't ideal - though it's to get some existing unit-tests to pass.  I'll merge this as is, though I'll raise some new PR's on this issue myself to get this changed to `string $name`